lib/facter/ec2.rb

```diff
@@ -46,8 +46,15 @@
     end
 
     setcode do
-      @querier.fetch
+      userdata = @querier.fetch
+      begin
+        userdata.encode(Encoding::UTF_8)
+      rescue EncodingError
```

Comment:
It looks like you could call normalize_string directly and catch the NormalizationError.